module/icp/algs/modes/gcm.c

```diff
@@ -88,7 +91,7 @@ gcm_mode_encrypt_contiguous_blocks(gcm_ctx_t *ctx, char *data, size_t length,
     void (*xor_block)(uint8_t *, uint8_t *))
 {
 #ifdef CAN_USE_GCM_ASM
-	if (ctx->gcm_use_avx == B_TRUE)
+	if (ctx->impl != GCM_IMPL_GENERIC)
```

Comment:
I think these tests need to be positive explicit tests ie `(ctx->impl == GCM_IMPL_AVX || ctx->impl == GCM_IMPL_AVX2)`, because I didn't see why `GCM_IMPL_PCLMULQDQ` wouldn't match. Turns out that doesn't exist though, and that thing is weirdly wired up a different way again. So hear my siiiiggghhh and then ignore me.